pkg/snmp/traps/listener_test.go

```diff
@@ -153,34 +154,33 @@ func TestListenerTrapsReceivedTelemetry(t *testing.T) {
 	config := Config{Port: serverPort, CommunityStrings: []string{"public"}, Namespace: "totoro"}
 	Configure(t, config)
 
-	packetOutChan := make(PacketsChannel)
+	packetOutChan := make(PacketsChannel, packetsChanSize)
 	trapListener, err := startSNMPTrapListener(config, mockSender, packetOutChan)
 	require.NoError(t, err)
 	defer trapListener.Stop()
 
 	sendTestV1GenericTrap(t, config, "public")
-	packet := receivePacket(t, trapListener, defaultTimeout) // Wait for packet
-	require.NotNil(t, packet)
+	_, err2 := receivePacket(t, trapListener, defaultTimeout) // Wait for packet
+	require.NoError(t, err2)
 	mockSender.AssertMetric(t, "Count", "datadog.snmp_traps.received", 1, "", []string{"snmp_device:127.0.0.1", "device_namespace:totoro", "snmp_version:1"})
 }
 
-func receivePacket(t *testing.T, listener *TrapListener, timeoutDuration time.Duration) *SnmpPacket {
+func receivePacket(t *testing.T, listener *TrapListener, timeoutDuration time.Duration) (*SnmpPacket, error) {
```

Comment:
TestServerV2BadCredentials expects receivePacket() to return nil https://github.com/DataDog/datadog-agent/blob/main/pkg/snmp/traps/listener_test.go#L93 indicating "invalid packet", however receivePacket() previously would also return nil in the case of a timeout, so changed it to return (packet, error) with different individual errors.